src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/model/ReviewLine.java

```diff
@@ -0,0 +1,217 @@
+     */
+    private final Map<String, String> properties;
+
+    public ReviewLine(Parent parent) {
+        this(parent, null);
+    }
+
+    public ReviewLine(final Parent parent, final String lineId) {
+        this.parent = Objects.requireNonNull(parent);
+        this.tokens = new ArrayList<>();
+        this.children = new ArrayList<>();
+        this.lineId = lineId;
+        this.properties = new HashMap<>();
+    }
+
+    public ReviewLine addToken(ReviewToken token) {
+        tokens.add(token);
+        return this;
+    }
+
+    public ReviewLine addToken(TokenKind tokenKind, String value) {
+        return addToken(tokenKind, value, Spacing.DEFAULT);
+    }
+
+    public ReviewLine addToken(TokenKind tokenKind, String value, Spacing spacing) {
+        return addToken(tokenKind, value, null, spacing);
+    }
+
+    public ReviewLine addToken(TokenKind tokenKind, String value, String navigateToId) {
+        return addToken(tokenKind, value, navigateToId, Spacing.DEFAULT);
+    }
+
+    public ReviewLine addToken(TokenKind tokenKind, String value, String navigateToId, Spacing spacing) {
+        return addToken(new ReviewToken(tokenKind, value, navigateToId).setSpacing(spacing));
+    }
+
+    public ReviewLine addSpace() {
+        return addToken(TokenKind.TEXT, " ", Spacing.NO_SPACE);
+    }
+
+    /**
+     * Returns the child that was passed in as the argument
+     * @param child
+     * @return
+     */
+    @Override
+    public ReviewLine addChildLine(ReviewLine child) {
+        children.add(child);
+        return child;
+    }
+
+    /**
+     * Add a review line with the given line ID
+     * @return the new review line
+     */
+    @Override
+    public ReviewLine addChildLine(final String lineId) {
+        return addChildLine(new ReviewLine(this, lineId));
+    }
+
+    @Override
+    public ReviewLine addChildLine() {
+        return addChildLine(new ReviewLine(this, null));
+    }
+
+    @Override
+    public List<ReviewLine> getChildren() {
+        return children;
+    }
+
+    public ReviewLine addProperty(String key, String value) {
+        properties.put(key, value);
+        return this;
+    }
+
+    public boolean hasProperty(String key) {
+        return properties.containsKey(key);
+    }
+
+    public String getProperty(String key) {
+        return properties.get(key);
+    }
+
+    public List<ReviewToken> getTokens() {
+        return tokens;
+    }
+
+    public ReviewLine addContextStartTokens() {
+        addToken(TokenKind.PUNCTUATION, "{");
+        return this;
+    }
+
+    public void addContextEndTokens() {
+        parent.addChildLine(new ReviewLine(this).addToken(TokenKind.PUNCTUATION, "}").setContextEndLine());
+    }
+
+    public ReviewLine setContextEndLine() {
+        isContextEndLine = true;
+        return this;
+    }
+
+    public String getLineId() {
+        return lineId;
+    }
+
+    public ReviewLine setRelatedToLine(ReviewLine relatedToLine) {
```

Comment:
I don't see any reference to `setRelatedToLine` method. If java shows any annotations for APIs like .NET attributes, then it needs to be marked as related to review line for the API. For e.g.

`@sampleAnnotaion
public void APIName()`

In above example, line `@sampleAnnotaion` should be marked as related to `public void APIName()`

Reply:
It is being used in JavaASTAnalyser [here](https://github.com/Azure/azure-sdk-tools/pull/9394/files#diff-4889c358554879f0c76804732fcd910d90c7c60060df2c5de7dfc52c9c3a7efcR1132) and [here](https://github.com/Azure/azure-sdk-tools/pull/9394/files#diff-4889c358554879f0c76804732fcd910d90c7c60060df2c5de7dfc52c9c3a7efcR1463)